.travis.yml

```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - "10"
+  - "14"
```

Comment:
We could also specify the version of Node.js with a .nvmrc file (https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#specifying-nodejs-versions-using-nvmrc). This would make it easier to enforce the version to use during development (tests won’t pass anymore on <14.13.0).

Reply:
That sounds fine as long as we aren't requiring people to use nvm. But I think having the tests fail on <14.13.0 is probably fine too. Why do the tests fail? Does the repl still work on older versions of node?

Reply:
The tests and the REPL need to be able to import CommonJS foreign modules with ES named imports, but this isn’t possible before v14.13.0:

```purs
module Example where

example = "Hello world"
```

```sh
> purs repl Example.purs 'bower_components/purescript-*/src/**/*.purs'
Compiling Example
PSCi, version 0.14.0
Type :? for help

> import Example
> example

```

```sh
> node
Welcome to Node.js v14.2.0.
Type ".help" for more information.
> import('./.psci_modules/index.js')
Promise { <pending> }
> (node:28607) UnhandledPromiseRejectionWarning:
export { cons, join, showArrayImpl, showCharImpl, showIntImpl, showNumberImpl, showStringImpl } from "./foreign.cjs";
         ^^^^
SyntaxError: The requested module './foreign.cjs' does not provide an export named 'cons'
    at ModuleJob._instantiate (internal/modules/esm/module_job.js:92:21)
    at async ModuleJob.run (internal/modules/esm/module_job.js:107:20)
    at async Loader.import (internal/modules/esm/loader.js:179:24)
```

I did not reconsider the version requirement when introducing ES wrappers for CommonJS foreign modules though, but we could import the default export of the CommonJS foreign module in the ES wrapper and re-export properties of the default export with export declarations:

```js
// foreign.cjs
"use strict";

exports.log = function (s) {
  return function () {
    console.log(s);
  };
};
```

```js
// foreign.js
import $foreign from "./foreign.cjs";

export var log = $foreign.log;
```

This has the advantage of working all the way back to Node.js v12.17.0 (earlier v12 versions need `--experimental-modules`) which is great news since support for v10 stops at the end of the month!

Reply:
This approach sounds good to me :+1:

Reply:
How far back is `--experimental-modules` understood?

Reply:
> Back in 2017, Node.js 8.9.0 shipped experimental support for ECMAScript modules, known for their import and export statements. This support was behind the flag --experimental-modules.

https://nodejs.medium.com/announcing-a-new-experimental-modules-1be8d2d6c2ff

Reply:
We’re really interested in the implementation of --experimental-modules that was shipped with v12.0.0, this flag is of no use to us in v10.0.0 and earlier.

Reply:
So if we can expect this approach to work when supplying `--experimental-modules` with any version of node.js >= v12.0.0, and if earlier versions of node are EOL, this sounds fine to me. I think we might want to check the version of `node` we are using in the repl and supply that flag where necessary. Perhaps we should test on node v12 as well as v14 for now as well?

Reply:
The REPL and the tests now throw on unsupported Node.js versions (<12.0.0) and provide the --experimental-module flag for versions >=12.0.0 && <12.7.0.

> Perhaps we should test on node v12 as well as v14 for now as well?

We don’t rely on the behaviour introduced in v14.13.0 anymore, so testing on v12.0.0 and the latest v12 might be enough?